tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,295 @@ int main(int argc, char **argv)
             }
         }
 
+        /* Test ChaCha20 boosting behaviour */
+        {
+            EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+
+            /* The server chose a (malformed) security policy where ChaCha20 boosting is toggled on but 
+             * the cipher preferences does not contain ChaCha20. In this case, ChaCha20 boosting behaviour is off.
+             */
+            {
+                static struct s2n_cipher_suite *cipher_suites_without_chacha20[] = {
+                    &s2n_tls13_aes_256_gcm_sha384,                  /* 0x13,0x02 */
+                    &s2n_tls13_aes_128_gcm_sha256,                  /* 0x13,0x01 */
+                };
+
+                const struct s2n_cipher_preferences cipher_preference_without_chacha20_and_boosting = {
+                    .count = s2n_array_len(cipher_suites_without_chacha20),
+                    .suites = cipher_suites_without_chacha20,
+                    .allow_chacha20_boosting = true
+                };
+
+                const struct s2n_security_policy test_no_chacha20_and_boosted = {
+                    .minimum_protocol_version = S2N_SSLv3,
+                    .cipher_preferences = &cipher_preference_without_chacha20_and_boosting,
+                    .kem_preferences = &kem_preferences_null,
+                    .signature_preferences = &s2n_signature_preferences_20201021,
+                    .ecc_preferences = &s2n_ecc_preferences_test_all,
+                };
+
+                uint8_t chacha20_boosted_wire[] = {
+                    TLS_CHACHA20_POLY1305_SHA256,
+                    TLS_AES_256_GCM_SHA384,
+                    TLS_AES_128_GCM_SHA256,
+                };
+
+                conn->security_policy_override = &test_no_chacha20_and_boosted;
+                conn->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                conn->client_protocol_version = S2N_TLS13;
+                conn->actual_protocol_version = S2N_TLS13;
+                conn->server_protocol_version = S2N_TLS13;
```

Comment:
It seems dangerous for all these tests to use the same connection if they're going to mutate that connection.